python/dolfinx/mesh.py

```diff
@@ -645,3 +645,26 @@ def create_unit_cube(comm: _MPI.Comm, nx: int, ny: int, nz: int, cell_type=CellT
     """
     return create_box(comm, [np.array([0.0, 0.0, 0.0]), np.array([1.0, 1.0, 1.0])],
                       [nx, ny, nz], cell_type, dtype, ghost_mode, partitioner)
+
+
+def entities_to_geometry(
+    mesh: Mesh, dim: int, entities: npt.NDArray[np.int32], orient: bool
+) -> npt.NDArray[np.int32]:
+    """Indices in the geometry data for each vertex of the given mesh entities.
+
+    Warning:
+        This function should not be used unless there is no alternative.
+        It may be removed in the future.
+
+    Args:
+        mesh: The mesh.
+        dim: Topological dimension of the entities of interest.
+        entities: Entity indices (local to the process) to determine the
+            vertex geometry indices for.
+        orient: If ``True``, in 3D, re-orients facets to have consistent
```

Comment:
What does this mean? (I know it straight from the C++ docs). @jorgensd?

Reply:
This is a flag only used by BEMPP (at least at its introduction) to ensure consistent orientation of each facet (to make all cell normals point outwards).

@mscroggs might be able to comment on the necessity of this at the current stage of BEMPP

Reply:
It it's still needed it needs an explanation of what is required for a consistent orientation, i.e. document how we related the normal vector to a facet to the vertex ordering. And does it apply for all facet shapes?

Reply:
This is still used by Bempp-cl, so I'm keen to keep it for now. I think we've only tested it for triangle facets, and I'd be keen to work towards removing this once Bempp-rs can do coupling without needing this.